pkg/tfvars/azure/azure.go

```diff
@@ -76,6 +78,12 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 		return nil, errors.Wrap(err, "could not determine Azure environment to use for Terraform")
 	}
 
+	imageURL := sources.ImageURL
+	_, ok := os.LookupEnv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE")
+	if sources.CloudName == azure.StackCloud && ok {
+		imageURL = sources.RHCOSImage
+	}
```

Comment:
Instead of doing the switch here, consider returning the value of `RHCOSImage` here https://github.com/openshift/installer/blob/master/pkg/asset/rhcos/image.go#L128-L137  (when it's azurestack). To me this is cleaner, and the OVERRIDE environment variable is already handled.

Reply:
This makes sense, I wasnt aware of that code path, i think i've added what you suggested